STAR-Delegation/CSR-template/template-schema.cddl

```diff
@@ -0,0 +1,80 @@
+csr-template-schema = {
+  ? keyTypes: [ 1* $keyType ]
+  ? subject: distinguishedName
+  ? extensions: extensions
+}
+
+mandatory-wildcard = "**"
+optional-wildcard = "*"
+wildcard = mandatory-wildcard / optional-wildcard
+
+; regtext matches all text strings but "*" and "**"
+regtext = text .regexp "([^\*].*)|([\*][^\*].*)|([\*][\*].+)"
+
+regtext-or-wildcard = regtext / wildcard
+
+distinguishedName = {
+  ? country: regtext-or-wildcard
+  ? stateOrProvince: regtext-or-wildcard
+  ? locality: regtext-or-wildcard
+  ? organization: regtext-or-wildcard
+  ? organizationalUnit: regtext-or-wildcard
+  ? emailAddress: regtext-or-wildcard
+  ? commonName: regtext-or-wildcard
+}
+
+$keyType /= rsaKeyType
+$keyType /= ecKeyType
+
+rsaKeyType = {
+  PublicKeyType: "RSA"
+  ? PublicKeyLength: $rsaKeySize
+  ? SignatureType: $signatureType / mandatory-wildcard
+}
+
+$rsaKeySize /= 2048
+$rsaKeySize /= 4096
+$rsaKeySize /= 8192
+
+$signatureType /= "sha256WithRSAEncryption"
+
+ecKeyType = {
+  PublicKeyType: "ecPublicKey"
+  ? Curve: $ecCurve
+  ? SignatureType: $ecSignatureType
+}
+
+$ecCurve /= "secp521r1"
+
+$ecSignatureType /= "ecdsa-with-SHA256"
+
+subjectaltname = {
+  ? DNS: [ 1* regtext ]
+  ? IP: [ 1* regtext-or-wildcard ]
```

Comment:
Just a fly-by thought. It makes no sense to specify IPs here: NDC's IP addresses can't be possibly validated by the IdO.

Reply:
I disagree (and ask that you add it back). In enterprise networks, the IdO might be able to validate IP addresses by querying cloud or asset management APIs.

Reply:
I'm still on the slightly puzzled side... by "validation" I mean the http-01 or tls-alpn-01 challenges that occur between the ACME CA and the IdO (RFC8738).  Are you envisaging the IdO handing over one of its IP address to the NDC after the challenge completes?

Reply:
You are right, and I misunderstood. Yes, let's drop IP SANs.